### PR TITLE
Add popup class nav

### PR DESCRIPTION
**Pull request by eireland**
Adds the class nav area with the view list by toggle and the spotlight toggle

### CODE REVIEW

**Comment by eireland** [comment]
@mklewandowski I think the is ready for review now

File: css/portal-dashboard/all-responses-popup/popup-class-nav.less [hunk 1]
```diff
@@ -0,0 +1,113 @@
+@import "../variables";
+
+.popupClassNav {
+  width: 240px;
```
Comment (mklewandowski) [inline css/portal-dashboard/all-responses-popup/popup-class-nav.less:4]: I think some bold fonts sneaked in.  Add this to a font-weight normal to fix in the popupClassNav class:
`font-weight: normal;`

File: js/components/portal-dashboard/all-responses-popup/popup-class-nav.tsx [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+  }
+  private renderViewListOptions() {
+    let listByStudentClasses: string = css.toggle + " " + css.listByStudents,
+      listByQuestionsClasses: string = css.toggle + " " + css.listByQuestions;
+
+    if (!this.state.inQuestionMode) {
+      listByStudentClasses += "  " + css.selected;
+    } else {
+      listByQuestionsClasses += " " + css.selected;
+    }
```
Comment (mklewandowski) [inline js/components/portal-dashboard/all-responses-popup/popup-class-nav.tsx:78]: You could convert this chunk to something slightly smaller with ternary:
```
    !this.state.inQuestionMode
      ? listByStudentClasses += " " + css.selected
      : listByQuestionsClasses += " " + css.selected;
```

Comment (mklewandowski) [reply js/components/portal-dashboard/all-responses-popup/popup-class-nav.tsx:78]: And if ambitious, you could probably add it to the let statements above to shrink everything down:
```
    const listByStudentClasses: string = `${css.toggle} ${css.listByStudents} `
        + (!this.state.inQuestionMode ? css.selected : "");
    const listByQuestionsClasses: string = `${css.toggle} ${css.listByQuestions} `
        + (this.state.inQuestionMode ? css.selected : "");
```

That would replace the let and the if statements.

File: js/components/portal-dashboard/all-responses-popup/popup-class-nav.tsx [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+  private handleStudentsToggleClick = () => {
+    this.setState({
+      inQuestionMode: false
+    });
+  }
+  private handleQuestionsToggleClick = () => {
+    this.setState({
+      inQuestionMode: true
+    });
+  }
```
Comment (mklewandowski) [inline js/components/portal-dashboard/all-responses-popup/popup-class-nav.tsx:120]: handleStudentsToggleClick and handleQuestionsToggleClick could be refactored into a single function that takes in a boolean, but isn't necessary.

File: js/components/portal-dashboard/all-responses-popup/popup-class-nav.tsx [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+  private handleSpotlightClick = () => {
+    const wasSpotlightOn = this.state.isSpotlightOn;
+    this.setState({
+      isSpotlightOn: !wasSpotlightOn
+    });
```
Comment (mklewandowski) [inline js/components/portal-dashboard/all-responses-popup/popup-class-nav.tsx:125]: You can also use `prevState`:
```
this.setState(prevState => ({
  isSpotlightOn: !prevState.isSpotlightOn
}));
```

File: js/components/portal-dashboard/all-responses-popup/popup-class-nav.tsx [hunk 5]
```diff
@@ -0,0 +1,137 @@
...
+    if (!wasSpotlightOn && !this.state.studentSelected) {
+      this.setState({ showDialog: true });
+    }
+  }
+
+  private closeShowDialog = (show: boolean) => {
+    this.setState({ showDialog: show });
+    this.setState({
+      isSpotlightOn: !this.state.isSpotlightOn
+    });
```
Comment (mklewandowski) [inline js/components/portal-dashboard/all-responses-popup/popup-class-nav.tsx:135]: You can condense these setStates into a single call:
```
    this.setState({
      isSpotlightOn: !this.state.isSpotlightOn,
      showDialog: show
    });
```

File: css/portal-dashboard/all-responses-popup/popup-class-nav.less (hunk 1 above)
Comment (eireland) [reply css/portal-dashboard/all-responses-popup/popup-class-nav.less:4]: Good catch! I've been assuming all these labels are bold.

File: js/components/portal-dashboard/all-responses-popup/popup-class-nav.tsx (hunk 3 above)
Comment (mklewandowski) [reply js/components/portal-dashboard/all-responses-popup/popup-class-nav.tsx:120]: Something like this:
```
  private setQuestionMode = (value: boolean) => () => {
    this.setState({
      inQuestionMode: value
    });
  }
```

And then, you can simply add JSX:
```
<div className={listByStudentClasses} data-cy="list-by-student-toggle" onClick={this.setQuestionMode(false)}>

<div className={listByQuestionsClasses} data-cy="list-by-questions-toggle" onClick={this.setQuestionMode(true)}>
```

**Comment by eireland** [comment]
Made the suggested changes. cleaned up the css logic in a few other places. The reuse of the ClassNav component shouldn't be too hard and I can do it later when we have a better idea of what the items to be passed into renderStudentSort from the popup view should be. Also I don't have the redux action/name for the items yet.

**Pull request merged**